scripts/combine_lightcurves.py

```diff
@@ -1,43 +1,93 @@
+#!/usr/bin/env python3
+
 import os
+import shutil
 import json
 import argparse
+from json.decoder import JSONDecodeError
 
-import glob
+from pocs.utils.google.storage import PanStorage
 
 
 def get_curves_for_pic(pic):
+    """Read light curve files and add all curves to array.
+
+    :param pic: the PIC id for the star to make a master light curve for
+    :return: an array of all the light curves for the given PIC
+    """
     curves = []
-    for filename in glob.iglob("data/LC/{}/**/*.json".format(pic), recursive=True):
-        with open(filename, 'r') as f:
-            c = json.load(f)
-            curves.append(c)
-    return curves
+    bucket_name = 'panoptes-simulated-data'
+    prefix = "LC/{}/".format(pic)
+    temp_dir = '{}/temp'.format(os.getenv('PANDIR', default='/var/panoptes'))
+
+    pan_storage = PanStorage(bucket=bucket_name)
+    blobs = pan_storage.list_remote(prefix)
+
+    for blob in blobs:
+        local_path = "{}/{}".format(temp_dir, blob.name)
+        path = pan_storage.download(blob.name, local_path=local_path)
+        with open(path, 'r') as f:
+            try:
+                curve = json.load(f)
+                curves.append(curve)
+            except JSONDecodeError:
+                print("Error: Object could not be decoded as JSON.")
+    return curves, temp_dir
 
 
 def combine_curves(curves):
+    """Flatten all the data points in the curves into one master curve array.
+
+    Each data point has a timestamp, exposure duration, RGB fluxes, and
+    RGB flux uncertainties. Eventually should add functionality for
+    overlapping exposures and be much less naive.
+    :param curves: an array of light curves to be combined
+    :return: a master light curve stored in a single array
+    """
     master = []
     for c in curves:
-        for t in c:
-            master.append(t)
+        for data_point in c:
+            master.append(data_point)
     return master
 
 
-def write_output(output, pic):
-    filename = 'output/{}.json'.format(pic)
-    os.makedirs(os.path.dirname(filename), exist_ok=True)
-    with open(filename, 'w') as fo:
-        json.dump(output, fo)
+def upload_output(pic, data, temp_dir):
```

Comment:
This is mostly I/O, so testing is probably less critical, but still a good idea if you have time.  You'll probably have to mock out the PanStorage stuff using something like https://docs.python.org/dev/library/unittest.mock.html